### PR TITLE
add material-dominant relationship support to blueprint matsets

### PR DESCRIPTION
**Pull request by xjrc**
The changes in this pull request address #512  by making the following changes:

- [x] Update the `blueprint::mesh::matset::verify` function to recognize material-dominant schemas.
- [x] Implement test cases to validate the correctness of the new `blueprint::mesh::matset::verify` functionality.
- [x] Update the ["Material Sets"](https://llnl-conduit.readthedocs.io/en/latest/blueprint_mesh.html#material-sets) documentation to include details for the new material-dominant option.
- [x] Update the `blueprint::mesh::examples::venn` regressions to all validate their results with `blueprint::mesh::verify`.

### CODE REVIEW

**Comment by xjrc** [comment]
@cyrush, @agcapps: This pull request is now ready for review, at your respective convenience.

File: src/docs/sphinx/blueprint_mesh.rst [hunk 1]
```diff
@@ -504,12 +516,85 @@ The following diagram illustrates a simple **multi-buffer** material set example
...
+              indices: [5, 3]
+            b:
+              values: [0, b0, b2, b1, 0]
+              indices: [1, 3, 2]
+
+
+Material Set Indexing Variants
+=================================
+
+Another dimension along which material sets can vary is in how their volume fractions/topological element association is established.
```
Comment (agcapps) [inline src/docs/sphinx/blueprint_mesh.rst:534]: I'd simplify and rephrase to "Material sets can also vary in how volume fractions are associated with topological elements."  I don't think we need to talk about "another dimension" since you haven't talked about the first dimension.

File: src/docs/sphinx/blueprint_mesh.rst [hunk 2]
```diff
@@ -504,12 +516,85 @@ The following diagram illustrates a simple **multi-buffer** material set example
...
+This dimension produces two additional material set variants: **element-dominant** association (element indices control volume fraction order) and **material-dominant** association (material indices control volume fraction order).
```
Comment (agcapps) [inline src/docs/sphinx/blueprint_mesh.rst:535]: Similar to the previous sentence, I'd rephrase to something like "Material sets can have **element-dominant** association (volume fractions for each element are grouped in the data buffer) or **material-dominant** association (volume fractions for eachmaterial are grouped in the data buffer)."

File: src/docs/sphinx/blueprint_mesh.rst [hunk 3]
```diff
@@ -504,12 +516,85 @@ The following diagram illustrates a simple **multi-buffer** material set example
...
+Both of these variants and their corresponding schemas are outlined in the subsections below.
+
+
+Element-Dominant Material Sets
+*********************************
+
+An **element-dominant** material set is one that orders its volume fraction data so that it matches the topological element ordering.
```
Comment (agcapps) [inline src/docs/sphinx/blueprint_mesh.rst:542]: Perhaps rephrase to "In an **element-dominant** material set, volume fraction data order matches topological element order."

**Comment by xjrc** [comment]
> Thank you, @xjrc . Much appreciated. I added some recommendations on phrasing in the .rst file---take them however you see fit.

Thanks Arlie; all good feedback! I'm going to incorporate your suggestions in my next push.

**Comment by xjrc** [comment]
@cyrush: Is this okay to merge? I'm keen to merge #566 in preparation for more changes in support of `o2mrelation` types and I'd like this to be finalized beforehand so as not to create too big a backlog of PRs dependent on one another.
